platform/newsfeed/src/main.rs

```diff
@@ -0,0 +1,291 @@
+use rdkafka::producer::{FutureProducer, FutureRecord};
+use rdkafka::util::Timeout;
+use rdkafka::ClientConfig;
+use rdkafka::Message;
+use regex::Regex;
+use scraper::{Html, Selector};
+use std::sync::Arc;
+use std::time::Duration;
+use url::Url;
+
+use newsfeed::{ClaudeMessage, ClaudeModel, ClaudeResponse, Prompt, Role, SentimentResponse};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::env;
+use tracing::{debug, error, info, instrument, warn};
+use uuid::Uuid;
+
+#[derive(Debug)]
+struct UpstashConfig {
+    bootstrap_server: String,
+    username: String,
+    password: String,
+}
+
+#[instrument]
+async fn create_consumer(upstash: Arc<UpstashConfig>) -> StreamConsumer {
+    ClientConfig::new()
+        .set("bootstrap.servers", &upstash.bootstrap_server)
+        .set("enable.partition.eof", "false")
+        .set("security.protocol", "SASL_SSL")
+        .set("sasl.mechanisms", "SCRAM-SHA-256")
+        .set("sasl.username", &upstash.username)
+        .set("sasl.password", &upstash.password)
+        .set("group.id", format!("chat-{}", Uuid::new_v4()))
+        .create()
+        .expect("Failed to create client")
+}
+
+#[instrument]
+async fn create_producer(upstash: Arc<UpstashConfig>) -> FutureProducer {
+    ClientConfig::new()
+        .set("bootstrap.servers", &upstash.bootstrap_server)
+        .set("security.protocol", "SASL_SSL")
+        .set("sasl.mechanisms", "SCRAM-SHA-256")
+        .set("sasl.username", &upstash.username)
+        .set("sasl.password", &upstash.password)
+        .create()
+        .expect("Failed to create producer")
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SentimentData {
+    // news_source: String,
+    confidence: i32,
+}
+
+async fn send_sentiment(
+    producer: &FutureProducer,
+    topic: &str,
+    sentiment_data: &SentimentResponse,
+) {
+    let payload = serde_json::to_string(&sentiment_data).unwrap();
+    let key = Uuid::new_v4().to_string();
+
+    let record = FutureRecord::to(topic).payload(&payload).key(&key);
+
+    match producer
+        .send(record, Timeout::After(Duration::from_secs(5)))
+        .await
+    {
+        Ok((partition, offset)) => {
+            debug!(
+                "Sent sentiment data to topic '{}', partition {}, offset {}",
+                topic, partition, offset
+            );
+        }
+        Err((e, _)) => {
+            error!("Failed to send sentiment data: {:?}", e);
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+enum RecordNamespace {
+    UpstashAvro,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RecordDatetime {
+    name: String,
+    #[serde(rename = "type")]
+    field_type: String,
+    doc: String,
+    #[serde(rename = "default")]
+    field_default: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NewsSource {
+    name: String,
+    #[serde(rename = "type")]
+    field_type: String,
+    doc: String,
+    field_default: Option<String>,
```

Comment:
Should there be a `serde` rename for this field?

Reply:
the current name is `field_default` from kafka, so doesn't need to be coerced to snake case